windows-server-container-tools/CleanupContainerHostNetworking/WindowsContainerNetworking-LoggingAndCleanupAide.ps1

```diff
@@ -0,0 +1,445 @@
+         }
+     }
+ }
+
+# -----------------------------------------------------------------------
+# This function collects logs to capture machine state
+# Parameters: -LogsPath <Path where logs should be saved>
+# Returns: _
+#
+# Information/logs collected by this function:
+#   - HNS Event log: c:\Windows\System32\winevt\Logs\Microsoft-Windows-Host-Network-Service-Admin.evtx
+#   - System event log: c:\Windows\System32\winevt\Logs\System.evtx
+#   - HNS.data file: c:\ProgramData\Microsoft\Windows\HNS\HNS.data
+#   - `ipconfig /allcompartments /all`
+#   - Info for each switch: `GetAllPolicies -switchName $switch.Name`
+#   - VFP info: Get-Service vfpext
+#   - Docker version: `docker -v`
+#   - Docker info: `docker info`
+#   - `Get-VMSwitch`
+#   - `Get-NetNat`
+#   - `Get-NetNatStaticMapping`
+#   - `Get-Service winnat`
+#   - `Get-Service mpssvc`
+#   - `Get-NetAdapter -IncludeHidden`
+#   - `Get-NetAdapterBinding -IncludeHidden`
+#   - `Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\hns`
+#   - `Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\vmsmp`
+# -----------------------------------------------------------------------
+function CopyAllLogs
+{
+    Param(
+      [string] $LogsPath
+    )
+
+    $LogsPath += "_$($script:namingSuffix)"
+
+    if (!(Test-Path $LogsPath))
+    {
+        mkdir $LogsPath
+    }
+
+    Copy-Item c:\Windows\System32\winevt\Logs\Microsoft-Windows-Host-Network-Service-Admin.evtx $LogsPath
+    Copy-Item c:\Windows\System32\winevt\Logs\System.evtx $LogsPath
+
+    $hns = Get-Service hns
+
+    try
+    {
+        if ($hns.Status -eq "Running")
+        {
+            net stop hns
+        }
+
+        if (Test-Path "C:\ProgramData\Microsoft\Windows\HNS\HNS.data")
+        {
+            Copy-Item c:\ProgramData\Microsoft\Windows\HNS\HNS.data $LogsPath
+        }
+    }
+    finally
+    {
+        if ($hns.Status -eq "Running")
+        {
+            net start hns
+        }
+    }
+
+    ipconfig /allcompartments /all > $LogsPath"\ipconfig.txt"
+
+    if (Test-Path "C:\Windows\System32\vfpctrl.exe")
+    {
+        $allSwitches = Get-VMSwitch
+        foreach ($switch in $allSwitches)
+        {
+            GetAllPolicies -switchName $switch.Name > $LogsPath"\SwitchPolicies_$($switch.Name).txt"
+        }
+
+        Get-Service vfpext | FL * > $LogsPath"\vfpext.txt"
+    }
+
+
+    # Get docker version
+    docker -v > $LogsPath"\docker_v.txt"
+
+    # Get docker info
+    docker info > $LogsPath"\docker_info.txt"
+
+    # Get list containing IDs of current container networks on host
+    $networks=docker network ls -q
+    # Initialize file for recording network info
+    "Number of container networks currently on this host: "+$networks.Length+"`\n" > $LogsPath"\docker_network_inspect.txt"
+    # Inspect each network
+    foreach($network in $networks)
+    {
+      $addMe = docker network inspect $network
+      Add-Content $LogsPath"\docker_network_inspect.txt" $addMe
```

Comment:
I guess I didn't realize > worked in PowerShell scripts. I typically use ` | Out-File -Append file.txt`